plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseDbLink.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.altibase.model;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.ext.generic.model.GenericStructContainer;
+import org.jkiss.dbeaver.model.DBPRefreshableObject;
+import org.jkiss.dbeaver.model.DBPScriptObject;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSObject;
+import org.jkiss.utils.CommonUtils;
+
+import java.sql.Timestamp;
+import java.util.Map;
+
+public class AltibaseDbLink extends AltibaseObject<GenericStructContainer> implements DBPScriptObject, DBPRefreshableObject {
+
+    protected String ddl;
+
+    private String userName; // null for public, user_id for private
+    private int linkId;
+    private int userMode; // 0:public, 1: private
+    private String remoteUserId;
+    private String targetName;
+    private int linkType; // Heterogeneous Link or Homogeneous Link: Only Heterogeneous since 6.5.1
+    private Timestamp created;
+    private Timestamp lastDdlTime;
+
+    protected AltibaseDbLink(GenericStructContainer parent, JDBCResultSet resultSet) {
+        super(parent, 
+                JDBCUtils.safeGetString(resultSet, "LINK_NAME"), 
+                JDBCUtils.safeGetLong(resultSet, "LINK_OID"),
+                true);
+
+        userName = JDBCUtils.safeGetString(resultSet, "USER_NAME");
+        linkId = JDBCUtils.safeGetInt(resultSet, "LINK_ID");
+        userMode = JDBCUtils.safeGetInt(resultSet, "USER_MODE");
+        remoteUserId = JDBCUtils.safeGetString(resultSet, "REMOTE_USER_ID");
+        linkType = JDBCUtils.safeGetInt(resultSet, "LINK_TYPE");
+        targetName = JDBCUtils.safeGetString(resultSet, "TARGET_NAME");
+        created = JDBCUtils.safeGetTimestamp(resultSet, "CREATED");
+        lastDdlTime = JDBCUtils.safeGetTimestamp(resultSet, "LAST_DDL_TIME");
+    }
+
+    @NotNull
```

Comment:
Nullability annotations are not required for primitive types, such as int, long, float, etc.

Reply:
Thanks, I'll remove the useless annotations from AltibaseDbLink, AltibaseDirectory, AltibaseJob, and AltibaseLibrary classes.